custom/icds_reports/tasks.py

```diff
@@ -1209,6 +1214,40 @@ def build_incentive_report(agg_date=None):
                 build_incentive_files.delay(location, agg_date, file_format, 3, location.parent.parent, location.parent)
 
 
+def is_launched_awc_incentive_report(launched_awc_ids):
+    awc_ids = launched_awc_ids.values_list('awc_id', flat=True)
+    awc_launched_count_from_aggregate = AggAwc.objects.filter(aggregation_level=5, awc_id__in=awc_ids).all()
+    csv_columns = ['awc_id', 'AwwIncentiveReport', 'AggAwc']
+    csv_data = []
+    for awc in launched_awc_ids:
+        awc_from_aggregate = awc_launched_count_from_aggregate.filter(awc_id=awc[0]).values_list('awc_id',
```

Comment:
This will do one db query per awc (currently around 500000). You should be able to do this as one query outside the loop, and then loop over the returned set. If both queries `AggAwc` and `AWWIncentive` order the returned values by awc_id, then you can loop over them at the same time comparing each entry as you get there.